lightning/src/ln/channel.rs

```diff
@@ -1935,7 +1973,12 @@ impl<ChanSigner: ChannelKeys> Channel<ChanSigner> {
 						},
 						&HTLCUpdateAwaitingACK::ClaimHTLC { ref payment_preimage, htlc_id, .. } => {
 							match self.get_update_fulfill_htlc(htlc_id, *payment_preimage) {
-								Ok(update_fulfill_msg_option) => update_fulfill_htlcs.push(update_fulfill_msg_option.0.unwrap()),
+								Ok((update_fulfill_msg_option, additional_monitor_update_opt)) => {
+									update_fulfill_htlcs.push(update_fulfill_msg_option.unwrap());
+									if let Some(mut additional_monitor_update) = additional_monitor_update_opt {
+										monitor_update.updates.append(&mut additional_monitor_update.updates);
```

Comment:
14d5b97

Shouldn't you decrement latest_monitor_update_id here too given it possible increase in get_update_fulfill_htlc ? (Or rather drop the decrement for next `send_commitment_no_status_check` as updates_id of monitor_update has already been incremented?)

Reply:
Hmm, AFAICT we should fix it later in the fn in any case. I don't see a control flow path where we don't, or were you just saying for belt-and-suspenders?

Reply:
Offline discussion turned up the issue here is that the comments all just refer to send_commitment_no_status_check (even though many other fns may increment it during the runtime here) but the code is correct. Comments have been updated.